### ISSUE TITLE
updater.addDownstreamChangeListener timing issues

### ISSUE THREAD

**Issue by michielbdejong** [linkeddata/rdflib.js#300]
My app opens 8 documents and wants to subscribe to changes for them. If I call `updater.addDownstreamChangeListener` immediately after `fetcher.load`, it fails. If I set a 100ms timeout, some of them succeed. If I set a 1000ms timeout, all of  them succeed.

The failure is always with error message `Server doies not support live updates thoughUpdates-Via :-(`. That error message also appears when you try to subscribe to a doc you haven't fetched yet, or when for instance you forgot to remove the `#this` fragment from the doc URL.

I'll try to dive into the code to find out how to make this more robust, and also to create a unit test that fails on master for this.

**Comment by jeswr** [comment]
Closing as part of the maintainer-signed-off triage (plan #824). This timing issue lives in the legacy Updates-Via websocket path, which modern Solid servers no longer serve; the live-update rework (including discovery) is tracked in #549. Please reopen with a fresh repro if you still hit this against a server that supports Updates-Via.

**Comment by bourgeoa** [comment]
Not sure this issue should be closed legacy web socket is not deprecated and apps may still use it.

**Comment by jeswr** [comment]
Fair point @bourgeoa — agreed, keeping this open. Updates-Via is still served in the wild (NSS/solidcommunity.net) and the code path is fully alive in v2.4.0 (`src/update-manager.ts` — `addDownstreamChangeListener` → `getUpdatesVia`, which reads the `updates-via` header out of the store's per-request fetch metadata; the reported race is that metadata not yet being queryable immediately after `load()` resolves, which also explains why a fragment-bearing URL fails the lookup entirely). Reclassifying this as an actionable bug on the legacy path rather than stale: the fix shape is either making `addDownstreamChangeListener` await/retry the metadata lookup or resolving it from the response object directly. #549 tracks the broader live-update rework but does not supersede fixing this. (Agent-assisted triage for @jeswr — thanks for the correction; plan #824's staleness call on this one was wrong.)
